### PR TITLE
Add support for predefined schedules to powercycleCentral plugin

### PR DESCRIPTION
**Pull request by simon-anz**
Per https://github.com/Innablr/revolver/issues/114

### CODE REVIEW

File: lib/config-schema.ts [hunk 1]
```diff
@@ -149,6 +149,7 @@ const ConfigSchema = z
           powercycleCentral: z
             .object({
               active: z.boolean(),
+              schedules: z.record(z.string(), z.string()).default({}),
```
Comment (alutman-innablr) [inline lib/config-schema.ts]: Shouldn't this be within the `config` block? It's also dependent on the parser selection. `strict` is just the only parser implementation atm.

If you want schedules shareable between configs, I'd be more inclined to introduce an account level setting (perhaps even as a generic pre-processing thing so you could define any field with a variable)

File: plugins/powercycleCentral.ts [hunk 2]
```diff
@@ -30,6 +31,7 @@ export default class PowerCycleCentralPlugin extends RevolverPlugin {
     this.scheduleTagName = this.pluginConfig.availabilityTag || 'Schedule';
     this.timezoneTagName = this.accountConfig.timezoneTag || 'Timezone';
     this.scheduleTagPriority = this.pluginConfig.availabilityTagPriority || 0;
+    this.predefinedSchedules = accountConfig.plugins[pluginName].schedules; // it's in a slightly strange place
```
Comment (alutman-innablr) [inline plugins/powercycleCentral.ts]: As per previous comment: `this.pluginConfig.schedules`

File: revolver-config-example.yaml [hunk 3]
```diff
@@ -64,11 +64,21 @@ defaults:
         - parser: strict
           # Look for this tag on resources to see if they have an explicit schedule set
           availabilityTag: Schedule
+          # List of "shorthand" schedules that can be referenced in the matchers
```
Comment (alutman-innablr) [inline revolver-config-example.yaml:67]: This example differs to the schema.

File: README.md [hunk 4]
```diff
@@ -318,12 +318,13 @@ a generic filter to match specific resources and what power schedule those resou
...
+| Option                  | Description                                                                | Allowed values                                             | Default    |
+| ----------------------- | -------------------------------------------------------------------------- | ---------------------------------------------------------- | ---------- |
+| parser                  | Set schedule interpretation format                                         | `strict`                                                   | `strict`   |
+| availabilityTag         | Set tag name for individual resource schedules                             | `string` AWS tag name                                      | `Schedule` |
+| availabilityTagPriority | Priority to set individually tagged schedules compared to the matchers.    | `number` >= 0                                              | `0`        |
+| schedules               | List of "shorthand" schedules that can be referenced in the matchers       | list of name:schedule See [strict](######strict) schedules | `[]`       |
```
Comment (alutman-innablr) [inline README.md]: Examples in readme differ to schema

**Review by alutman-innablr — CHANGES_REQUESTED** [review]

File: lib/config-schema.ts (hunk 1 above)
Comment (lyricnz) [reply lib/config-schema.ts:152]: Moving to under config. KISS on the reuse, for now.

Comment (simon-anz) [reply lib/config-schema.ts]: I came to the same conclusion while writing the code, hence the README.md and revolver-config-example.yaml documenting same. Updated the schema now, and renamed it to predefinedSchedules for clarity.

**Review by alutman-innablr — APPROVED** [review]

**Pull request merged**